posthog/caching/insight_caching_state.py

```diff
@@ -60,20 +78,16 @@ def upsert(
     return caching_state
 
 
-def fetch_states_in_need_of_updating():
-    return InsightCachingState.objects.raw(
-        """
-        SELECT *
-        FROM posthog_insightcachingstate
-        WHERE target_cache_age_seconds IS NOT NULL
-          AND refresh_attempt < %(max_attempts)s
-          AND (
-            last_refresh IS NULL OR
-            last_refresh < %(timestamp)s - target_cache_age_seconds * interval '1' second
-          )
-        """,
-        {"max_attempts": MAX_ATTEMPTS, "timestamp": now()},
-    )
+def sync_insight_caching_state(team_id: int, insight_id: Optional[int] = None, dashboard_tile_id: Optional[int] = None):
+    try:
+        team = Team.objects.get(pk=team_id)
+        if dashboard_tile_id is not None:
+            upsert(team, DashboardTile.objects.get(pk=dashboard_tile_id))
+        elif insight_id is not None:
+            upsert(team, Insight.objects.get(pk=insight_id))
+    except Exception as err:
+        # This is a best-effort kind synchronization, safe to ignore errors
+        logger.warn("Failed to sync InsightCachingState, ignoring", exception=err)
```

Comment:
Fair. I was thinking that sentry will have that context, but you're right that logger does not!

~~To avoid rebase hell, will follow up with this.~~ Fixing since I need to fix another thing anyways.